src/pages/DatasetCatalog.js

```diff
@@ -251,24 +250,35 @@ export default function DatasetCatalog(props) {
     return true;
   };
 
+  const filteredAndPaginatedDatasetList = () => {
+    return datasetList.filter(searchTable(searchDulText)).slice((currentPage - 1) * pageSize, currentPage * pageSize);
+  };
+
+  const allOnPageSelected = () => {
+    return filteredAndPaginatedDatasetList().every((row) => {
+      return row.checked || (!row.active);
+    });
+  };
+
   const selectAll = (e) => {
```

Comment:
nit: change name to reflect that only the visible datasets are being selected (ex: selectAllOnPage)